core/server/worker/src/main/java/alluxio/worker/block/reviewer/ProbabilisticReviewer.java

```diff
@@ -0,0 +1,56 @@
+package alluxio.worker.block.reviewer;
+
+import alluxio.conf.PropertyKey;
+import alluxio.conf.ServerConfiguration;
+import alluxio.worker.block.meta.StorageDirView;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Random;
+
+/**
+ * Reviews a block allocation, and probably reject a bad block allocation.
+ * The more full a StorageDir, the higher the chance that this reviewer decides
+ * not to use it for a new block.
+ * */
+public class ProbabilisticReviewer implements Reviewer {
+    private static final Logger LOG = LoggerFactory.getLogger(ProbabilisticReviewer.class);
+
+    // TODO(jiacheng): Make this a property
+    private static final double CUTOFF = 0.1;
+    private static final Random RANDOM = new Random();
+
+    public ProbabilisticReviewer() {}
+
+    public boolean reviewAllocation(StorageDirView dirView) {
+      long availableBytes = dirView.getAvailableBytes();
+      long capacityBytes = dirView.getCapacityBytes();
+      long blockSize = ServerConfiguration.global().getBytes(PropertyKey.USER_BLOCK_SIZE_BYTES_DEFAULT);
+      long cutoffSize = (long) Math.floor(CUTOFF * capacityBytes);
+      // Rules:
+      // 1. If more than CUTOFF left, we use this tier. Prob=100%
+      // 2. If less that block size left, we ignore this tier. Prob=0%
+      // 3. If in the middle, the probability is linear to the space left, the less space the lower.
```

Comment:
@gpang Does this heuristic of probability make sense to you? Do you think instead of a linear function, something quadratic or logistic make more sense to make the probability drop quicker?

Reply:
I think it makes sense, but we could just make that factor a parameter. Something like 
```
((cutoff - free) / cutoff) ^ alpha
```

if `alpha` is 1, it is linear, and the user can adjust the curve by adjusting `alpha`. Maybe there are nicer curves out there, but this is advanced and may not be necessary.

Reply:
Agreed there are nicer curves but may not be necessary. After all the difference would be just like 1 block. I think we can just start with a linear function. I will try out higher rank functions like quadratic to the space left in the near future, but not in this PR.